### ISSUE TITLE
Parser gap: adapter composition rejects valid Zsh when alternate if, while short form and try/always appear in that order

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#337]
`TestAdapterCompositionAllFeaturesTogether` fails on roughly 1 in 7 runs on `main`, and the underlying cause is a real parser false positive, not just test noise.

Measured on `main` (`fcba451`), 60 uncached runs of that test alone:

```
flaked 9 / 60   (15%)
```

The test builds one file from `adapterSnippets`, which is a map, so Go randomizes the order on every run. The comment calls that "deliberate extra coverage ... since no ordering may matter" — but an ordering does matter, so the randomization turns a genuine defect into an intermittent failure.

## The defect

Delta-debugging a failing permutation shrinks it to three snippets:

```zsh
#!/usr/bin/env zsh
if (( 1 )) { x=1 }
while (( i < 3 )) (( i++ ))
{ true } always { true }
```

zsh accepts this:

```
$ zsh -f -n min3.zsh ; echo $?
0
```

zsh-lint rejects it:

```
$ zsh-lint min3.zsh
min3.zsh:4:1: `;` can only immediately follow a statement
exit=1
```

All three lines are individually valid and all 552 ordered *pairs* of snippets parse. The failure needs all three together, in that order: the alternate brace-form `if`, then an undelimited `while` short form, then a `try`/`always` block. Dropping any one of them makes it parse, so this is a composition defect in the adapter chain rather than a defect in any single adapter.

Randomized search over full-length permutations gives 13 failures in 4000 (0.3%), consistent with the observed per-run flake rate once you account for how many orderings put those three in the wrong relative position.

## Scope

Pre-existing on `main`, reproduced with a binary built from `origin/main` at `fcba451`. Not introduced by #326, #328 or #329, and unrelated to the CI change in #336 — that PR touches only `.github/workflows/zsh-n.yml` and `internal/workflowcontract/`.

## Suggested fix

Two separate things:

1. Fix the composition defect so the three-line file above parses.
2. Make the test deterministic. Iterating a map means a failure is unreproducible from the log alone: the printed source is the only record of the order, and a rerun usually passes, which invites treating a real defect as flake. Sorting the snippet names and asserting every permutation is too expensive at n=24, but sorting for the main assertion and adding a separate seeded randomized test that prints its seed on failure gives reproducibility and keeps the extra coverage.
